core/contractsdk/java/example/erc20/src/main/java/com/baidu/xuper/Erc20.java

```diff
@@ -0,0 +1,318 @@
+        BigInteger callerBalanceNow = callerBalance.add(increaseSupply);
+
+        ctx.putObject(balanceKey.getBytes(), callerBalanceNow.toByteArray());
+
+        return Response.ok("mint success".getBytes());
+    }
+
+    @ContractMethod
+    public Response totalSupply(Context ctx) {
+        byte[] value = ctx.getObject(TOTALSUPPLY.getBytes());
+        if (value == null) {
+            return Response.error("key TOTALSUPPLY not found)");
+        }
+        BigInteger totalSupply = new BigInteger(value);
+
+        return Response.ok(totalSupply.toString().getBytes());
+    }
+
+    @ContractMethod
+    public Response balance(Context ctx) {
+        byte[] accountByte = ctx.args().get("account");
+        if (accountByte == null) {
+            return Response.error("missing account");
+        }
+        String account = new String(accountByte);
+
+        String balanceKey = BALANCEPRE + account;
+        byte[] value = ctx.getObject(balanceKey.getBytes());
+        if (value == null) {
+            return Response.error("key " + account + " not found");
+        }
+        BigInteger balance = new BigInteger(value);
+
+        return Response.ok(balance.toString().getBytes());
+    }
+
+    @ContractMethod
+    public Response allowance(Context ctx) {
+        byte[] fromByte = ctx.args().get("from");
+        if (fromByte == null) {
+            return Response.error("missing from");
+        }
+        String from = new String(fromByte);
+
+        byte[] toByte = ctx.args().get("to");
+        if (toByte == null) {
+            return Response.error("missing to");
+        }
+        String to = new String(toByte);
+
+        String allowanceKey = ALLOWANCEPRE + from + "_" + to;
+        byte[] value = ctx.getObject(allowanceKey.getBytes());
+        if (value == null) {
+            return Response.error("key " + allowanceKey + " not found");
+        }
+        BigInteger allowance = new BigInteger(value);
+
+        return Response.ok(allowance.toString().getBytes());
+    }
+
+    @ContractMethod
+    public Response owner(Context ctx) {
+        byte[] ownerByte = ctx.getObject(MASTERPRE.getBytes());
+        if (ownerByte == null) {
+            return Response.error("no owner found");
+        }
+
+        return Response.ok(ownerByte);
+    }
+
+    @ContractMethod
+    public Response transfer(Context ctx) {
+        String from = ctx.caller();
+        if (from.isEmpty()){
+            return Response.error("missing from");
+        }
+
+        byte[] toByte = ctx.args().get("to");
+        if (toByte == null) {
+            return Response.error("missing to");
+        }
+        String to = new String(toByte);
+
+        byte[] tokenByte = ctx.args().get("token");
+        if (tokenByte == null) {
+            return Response.error("missing token");
+        }
+        String tokenAmountStr = new String(tokenByte);
+        BigInteger tokenAmount = new BigInteger(tokenAmountStr);
```

Comment:
What if the token is negative?

Reply:
It's fixed. Token amount could not be negative now.